### ISSUE TITLE
train on custom dataset?

### ISSUE THREAD

**Issue by daniiki** [wvangansbeke/Unsupervised-Classification#34]
Hi, thanks for this great project. 
I have a dataset with images that show a dotted font on a white background. These images have different kind of artifacts like pollution, scratches, ... often there are also several artifacts in one image. I want to cluster the dataset according to the different kind of artifacts. Do you think this is possible with this code?

**Comment by wvangansbeke** [comment]
Hi @daniiki,

I think you can always try to solve the pretext task and check the quality of the nearest neighbors. If they are ok you should be able to obtain good clustering accuracy. Personally I think it will be quite difficult based on the dataset you described. However, you won't know, if you don't try it. Have a look at issues #8 and #19 for your custom dataset.

Hope this helps.

**Comment by daniiki** [comment]
Hi :) thanks for the very fast response! 
I had a look at the _getitem_ function of cifar.py and stl.py in order to write a function for loading my own dataset. But there it seems like you need labels/targets for every image? However, my dataset is not labeled. Is it not possible to use the code without any labels?

**Comment by wvangansbeke** [comment]
Yes indeed. You will have to remove the parts where we load and use the labels for evaluation. You'll have to write the dataloader yourself anyway. I don't think it's much work though.

**Comment by daniiki** [comment]
Now, I wrote a custom Dataloader class and I tried rewriting the rest of the code, but I do not understand how to rewrite the simclr.py file in order to run the pretext task without validation. 
I have problems with these line of codes: 
` # Memory Bank
    print(colored('Build MemoryBank', 'blue'))
    base_dataset = get_train_dataset(p, val_transforms, split='train') # Dataset w/o augs for knn eval
    base_dataloader = get_val_dataloader(p, base_dataset) 
    memory_bank_base = MemoryBank(len(base_dataset), 
                                p['model_kwargs']['features_dim'],
                                p['num_classes'], p['criterion_kwargs']['temperature'])
    memory_bank_base.cuda()
    memory_bank_val = MemoryBank(len(val_dataset),
                                p['model_kwargs']['features_dim'],
                                p['num_classes'], p['criterion_kwargs']['temperature'])
    memory_bank_val.cuda()`

Actually I want to comment out the base_dataset and base_dataloader line, because I have no val_transforms, but base_dataloader is also used during training: 

`   # Fill memory bank
        print('Fill memory bank for kNN...')
        fill_memory_bank(base_dataloader, model, memory_bank_base)

        # Evaluate (To monitor progress - Not for validation)
        print('Evaluate ...')
        top1 = contrastive_evaluate(val_dataloader, model, memory_bank_base)
        print('Result of kNN evaluation is %.2f' %(top1)) 
 `
Do I need this during training? 
Sorry for the many questions! :( But I'm a little bit confused right now

**Comment by wvangansbeke** [comment]
Not really. This is only necessary for validation. To evaluate with kNN the complete training set is used (i.e. memory_bank_base).

**Comment by wvangansbeke** [comment]
If you have any more issues, feel free to ask more questions. I'm closing this for now.
